modules/phase_field/src/userobjects/EBSDReader.C

```diff
@@ -322,20 +322,36 @@ EBSDReader::buildNodeWeightMaps()
     _node_to_phase_weight_map[node_id].resize(getPhaseNum(), 0.0);
 
     // Loop through element indices associated with the current node and record weighted eta value in new map
-    unsigned int n_elems = node_to_elem_map[node_id].size();  // n_elems can range from 1 to 4 for 2D and 1 to 8 for 3D problems
+    const unsigned int n_elems = node_to_elem_map[node_id].size();
+    unsigned int used_elems = 0;
 
+    // count number of non-ancestor elements touching this node
     for (unsigned int ne = 0; ne < n_elems; ++ne)
     {
-      // Current element index
-      unsigned int elem_id = node_to_elem_map[node_id][ne];
+      // Current element index and elment pointer
+      const unsigned int elem_id = node_to_elem_map[node_id][ne];
+      const Elem * elem = mesh.elem(elem_id);
 
-      // Retrieve EBSD grain number for the current element index
+      // skip elements that are not the most refined state
+      if (!elem->ancestor()) used_elems++;
```

Comment:
OK. But why would an unrefined mesh have ancestor equal true? And that
seems to be what is in libmesh...

On Mon, Dec 7, 2015, 8:29 PM Derek Gaston <notifications@github.com> wrote:

> In modules/phase_field/src/userobjects/EBSDReader.C
> <https://github.com/idaholab/moose/pull/6032#discussion_r46911715>:
>
> >
> > -      // Retrieve EBSD grain number for the current element index
> > +      // skip elements that are not the most refined state
> > +      if (!elem->ancestor()) used_elems++;
>
> This isn't the right check... because it won't do the right thing on a
> level 0 (i.e., unrefined) mesh. The correct check is elem->active()
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/idaholab/moose/pull/6032/files#r46911715>.
>